sklearn/metrics/cluster/_supervised.py

```diff
@@ -1022,16 +1031,25 @@ def normalized_mutual_info_score(
     clusters = np.unique(labels_pred)
 
     # Special limit cases: no clustering since the data is not split.
+    # It corresponds to both labellings having zero entropy.
     # This is a perfect match hence return 1.0.
     if (
         classes.shape[0] == clusters.shape[0] == 1
         or classes.shape[0] == clusters.shape[0] == 0
     ):
         return 1.0
+
     contingency = contingency_matrix(labels_true, labels_pred, sparse=True)
     contingency = contingency.astype(np.float64, **_astype_copy_false(contingency))
     # Calculate the MI for the two clusterings
     mi = mutual_info_score(labels_true, labels_pred, contingency=contingency)
+
+    # At this point mi = 0 can't be a perfect match (the special case of a single
+    # cluster has been dealt with before). Hence, if mi = 0, the nmi must be 0 whatever
+    # the normalization.
+    if mi == 0:
```

Comment:
Usually comparing to floats is a bad practice, I think this is okay as we are trying to check the special case returned by `mutual_info_score`.

Reply:
Yes in this specific case it's acceptable because we want to check that the value is exactly 0 which doesn't come from some computation but from a hard coded 0.